docs/tutorial/code/expressjs/task.yaml

```diff
@@ -0,0 +1,257 @@
+###########################################
+# IMPORTANT
+# Comments matter!
+# The docs use the wrapping comments as
+# markers for including said instructions
+# as snippets in the docs.
+###########################################
+summary: Getting started with ExpressJS tutorial
+
+kill-timeout: 180m
+
+prepare: |
+  juju destroy-model expressjs-hello-world \
+  --destroy-storage --no-prompt --force || true
+  juju destroy-controller --force --destroy-all-models \
+  --destroy-storage --no-prompt --no-wait dev-controller || true
+
+environment:
+
+execute: |
+  # Move everything to $HOME so that Juju deployment works
+  mv *.yaml *.js *.sh $HOME
+  cd $HOME
+
+  # Don't use the staging store for this test
+  unset CHARMCRAFT_STORE_API_URL
+  unset CHARMCRAFT_UPLOAD_URL
+  unset CHARMCRAFT_REGISTRY_URL
+
+  # MicroK8s config setup
+  microk8s status --wait-ready --timeout 120
+  microk8s enable hostpath-storage
+  microk8s enable registry
+  microk8s enable ingress
+
+  # Bootstrap controller
+  juju bootstrap microk8s dev-controller --config bootstrap-timeout=240
+
+  # [docs:create-working-dir]
+  mkdir expressjs-hello-world
+  cd expressjs-hello-world
+  # [docs:create-working-dir-end]
+
+  cd ..
+  mv *.yaml *.js *.sh $HOME/expressjs-hello-world
+  cd expressjs-hello-world
+
+  # [docs:install-init-expressjs]
+  sudo apt update -y && sudo apt install npm -y
+  sudo npm install -g express-generator@4
+  express app
+  cd app
+  # [docs:install-init-expressjs-end]
+
+  # [docs:build-expressjs]
+  npm install
+  # [docs:build-expressjs-end]
+
+  npm start >npmoutput.txt 2>&1  </dev/null &
+
+  retry -n 5 --wait 2 curl --fail localhost:3000
+
+  # [docs:curl-expressjs]
+  curl localhost:3000
+  # [docs:curl-expressjs-end]
+
+  kill $!
+  pkill --signal SIGINT node
+
+  # [docs:create-rockcraft-yaml]
+  cd ..
+  rockcraft init --profile expressjs-framework
+  # [docs:create-rockcraft-yaml-end]
+
+  sed -i "s/name: .*/name: expressjs-hello-world/g" rockcraft.yaml
+  sed -i "s/amd64/$(dpkg --print-architecture)/g" rockcraft.yaml
+
+  # [docs:pack]
+  ROCKCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS=true rockcraft pack
+  # [docs:pack-end]
+
+  # [docs:skopeo-copy]
+  rockcraft.skopeo copy \
+    --insecure-policy \
+    --dest-tls-verify=false \
+    oci-archive:expressjs-hello-world_0.1_$(dpkg --print-architecture).rock \
+    docker://localhost:32000/expressjs-hello-world:0.1
+  # [docs:skopeo-copy-end]
+
+  # [docs:create-charm-dir]
+  mkdir charm
+  cd charm
+  # [docs:create-charm-dir-end]
+
+  # [docs:charm-init]
+  charmcraft init --profile expressjs-framework --name expressjs-hello-world
+  # [docs:charm-init-end]
+
+  # update platforms in charmcraft.yaml file
+  sed -i "s/amd64/$(dpkg --print-architecture)/g" charmcraft.yaml
+
+  # [docs:charm-pack]
+  CHARMCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS=true charmcraft pack
```

Comment:
I count six instances of the env var. Can we instead ask the user to set it in their environment, so we don't have to repeat ourselves so often?